RA-10-fragments-data-transfer

DenisGubernatorov opened the pull request:
1) Протянул передачу данных о выбранной категории между фрагментами

DenisGubernatorov committed:
RA-10-fragments-data-transfer

1) Протянул передачу данных о выбранной категории между фрагментами

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -34,23 +34,37 @@ class CategoriesListFragment : Fragment() {
+        val bundle = Bundle().apply {
+            putInt(RecipesListFragment.ARG_CATEGORY_ID, categoryId)
+            putString(RecipesListFragment.ARG_CATEGORY_NAME, category.title)
+            putString(RecipesListFragment.ARG_CATEGORY_IMAGE_URL, category.imageUrl)
```
Можно использовать более свежий и красивый подход, который нам доступен благодаря ktx библиотеке.
val bundle = bundleOf(ARG_CATEGORY_ID to categoryId, ...)

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipesListFragment.kt:
```diff
@@ -8,17 +8,30 @@ import androidx.fragment.app.Fragment
         _binding = RecipesListFragmentBinding.inflate(inflater)
+        categoryId = requireArguments().getInt(ARG_CATEGORY_ID)
+        categoryName = requireArguments().getString(ARG_CATEGORY_NAME)
+        categoryImageUrl = requireArguments().getString(ARG_CATEGORY_IMAGE_URL)
```
Давай стелабем код почище и вынесем arguments в with(...){}

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipesListFragment.kt:
```diff
@@ -8,17 +8,30 @@ import androidx.fragment.app.Fragment
         savedInstanceState: Bundle?
     ): View {
         _binding = RecipesListFragmentBinding.inflate(inflater)
+        categoryId = requireArguments().getInt(ARG_CATEGORY_ID)
```
Предлагаю для красоты использовать проперти arguments. requireArguments(), если в аргументах будет null, не возвращает null, а просто выбрасывает exception. Поэтому для arguments можно применить let {} и внутри этого безопасного блока инициализировать сразу три переменных.

DenisGubernatorov committed:
RA-10-fragments-data-transfer

1) Установка параметров для bundle переведено на bundleOf
2) Получение параметров из bundle переведено на использование let

DenisGubernatorov commented:
Получается что можно использовать или with или let. Решил использовать let, так как там не выбросит ошибку если arguments будет null.

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipesListFragment.kt:
```diff
@@ -29,9 +29,12 @@ class RecipesListFragment : Fragment() {
-        categoryName = requireArguments().getString(ARG_CATEGORY_NAME)
-        categoryImageUrl = requireArguments().getString(ARG_CATEGORY_IMAGE_URL)
+
+        requireArguments().let {
```
Да, с let правильно. Но я выше писал, что лучше обращаться к arguments вместо requireArguments(). А чтоб обходить ситуацию с null, надо добавить вопросительный знак и получится arguments?.let{...}

aGroupoid reviewed (approved):
По заданию всё верно, молодец! Поправь в следующем Пр один коммент
